x-pack/plugins/enterprise_search/public/applications/shared/hidden_text/hidden_text.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, ReactElement } from 'react';
+
+interface ChildrenProps {
+  toggle: () => void;
+  isHidden: boolean;
+  hiddenText: string;
+}
+
+interface Props {
+  text: string;
+  children(props: ChildrenProps): ReactElement;
+}
+
+export const HiddenText: React.FC<Props> = ({ text, children }) => {
+  const [isHidden, toggleIsHidden] = useState(true);
+  const hiddenText = isHidden ? text.replace(/./g, '*') : text;
+
+  return children({
+    hiddenText,
+    isHidden,
+    toggle: () => toggleIsHidden(!isHidden),
+  });
```

Comment:
I totally get what we were going for here in terms of matching EuiCopy's renderProp functionality. However, here's my question - do we ever intend on using hidden text without the accompanying "eye" button icon and clicking it to reveal text? If not, IMO we should just bake that into the HiddenText component. We don't need to be as general as EUI is with their components & can be a bit more prescriptive since we know the limitations & usage of our app.

The primary reasons I'd like to do so is

1. Ease of dev use, i.e. simply doing `<HiddenText text={text} />` (and having it manage its own internal state) vs `<HiddenText text={text}>({ whateverRenderProps }) => etc... </HiddenText>`

2. Accessibility - I want to ensure we bake accessibility into this component so that devs don't have to think about it more than once. Right now screen readers are going to read the hidden text either "Starstarstar" (or "Bulletbulletbullet" in the case of my above suggestion) which isn't great. I'd prefer to wrap the text a descriptive `aria-label` that says something along the lines of "Hidden text".

I'll make some screen reader suggestions above on how to manage aria-labels & text!

Reply:
Actually - after playing around with this a bit more, I think I totally retract my statement about making this a generic component vs a render prop. The aria-labels on the show/hide icon is what sold me on leaving it as more flexible. Components can set their own "Hide XYZ" / "Show XYZ" labels that way which is far more helpful screen-reader-wise.

Collapsing this comment, will leave some more accessibility suggestions above!

Reply:
Thank you for the thoughtful comments.